Library/Homebrew/dev-cmd/extract.rb

```diff
@@ -132,7 +132,15 @@ def extract
       loop do
         rev = rev.nil? ? "HEAD" : "#{rev}~1"
         rev, (path,) = Git.last_revision_commit_of_files(repo, pattern, before_commit: rev)
-        odie "Could not find #{name}! The formula or version may not have existed." if rev.nil?
+        if rev.nil? && (source_tap.path/".git/shallow").exist?
+          opoo <<~EOS
+            #{source_tap} is shallow clone. Run
+              git -C $(brew --repository homebrew/homebrew-core) fetch --unshallow
+              and retry.
+          EOS
+        elsif rev.nil?
+          odie "Could not find #{name}! The formula or version may not have existed."
+        end
```

Comment:
```suggestion
        if rev.nil? && source_tap.shallow?
          opoo <<~EOS
            #{source_tap} is shallow clone! Retry after running:
              git -C "$(#{source_tap.path})" fetch --unshallow
          EOS
        elsif rev.nil?
          odie "Could not find #{name}! The formula or version may not have existed."
        end
```

Reply:
On it! @MikeMcQuaid  😄

Reply:
@vidusheeamoli Thanks! Note you can commit straight from the GitHub UI if desired 😎